src/builder.ts

```diff
@@ -565,25 +594,98 @@ export class SchemaBuilder {
       return true
     }
 
-    const message = `${typeName}.${fieldName} is referencing a type "${outputType}" that is not defined in your GraphQL schema.`
+    const message = `Your GraphQL \`${typeName}\` object definition is projecting a field \`${fieldName}\` with \`${outputType}\` as output type, but \`${outputType}\` is not defined in your GraphQL Schema`
```

Comment:
The more concrete the better I think. Every mental hope removed in error messages helps. Our code can figure out _which_ kinds of output type are in question.